packages/device_info_plus/device_info_plus/macos/Classes/SystemUUID.swift

```diff
@@ -4,7 +4,7 @@ public struct SystemUUID {
     
     public static func getSystemUUID() -> String? {
         let dev = IOServiceMatching("IOPlatformExpertDevice")
-        let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMasterPortDefault, dev)
+        let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMainPortDefault, dev)
```

Comment:
Your current implementation will break devices running older versions of macOS (as seen in the [tests](https://github.com/fluttercommunity/plus_plugins/actions/runs/6842421579/job/18603873438?pr=2377#step:6:45)).
Instead, use something like this (not sure if it works directly, you might need to fix syntax errors):

```suggestion
        if #available(macOS 12, *) {
            let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMainPortDefault, dev)
        } else {
            let platformExpert: io_service_t = IOServiceGetMatchingService(kIOMasterPortDefault, dev)
        }
```

Reply:
> Your current implementation will break devices running older versions of macOS (as seen in the [tests](https://github.com/fluttercommunity/plus_plugins/actions/runs/6842421579/job/18603873438?pr=2377#step:6:45)). Instead, use something like this (not sure if it works directly, you might need to fix syntax errors):

Yes that is correct.